### PR TITLE
update hyper

### PR DESCRIPTION
**Pull request by mattjperez**
verifying everything works the same. wasn't prompted for other code changes

### CODE REVIEW

**Review by acmorrow — CHANGES_REQUESTED** [review]

File: Cargo.lock [hunk 1]
```diff
@@ -2061,7 +2061,7 @@ dependencies = [
  "httpdate",
  "itoa",
  "pin-project-lite",
- "socket2 0.5.7",
+ "socket2 0.4.10",
```
[the remark this answers is not in the document]
Comment (mattjperez) [reply Cargo.lock:2064]: weird. that's what was resolved using `cargo update hyper`.

checking the `cargo-tree`, it doesn't appear to have been a dependency for anything else. don't know why that got backed down. 

Looking into `socket2` they support two versions at any given time, right now, v4 and v5, on different branches but neither are the tagged release on github.

bumping back up and testing.

[the remark this answers is not in the document]
Comment (mattjperez) [reply Cargo.lock:2064]: @acmorrow ok I think I have an idea of why it resolved to `0.4.10`.

I did a full `cargo update` and found that there's a nested dependency on `socket2@0.4.10` for `async-std-openssl@0.6.3` which we have as a direct workspace dependency.

The nested dependency goes `async-std-openssl@0.6.3 -> async-std@1.12.0 -> async-io@1.13.0 -> socket2@0.4.10`

not sure why it got downgraded when running the upgrade, but this is why it's in our tree at all.

File: Cargo.toml [hunk 2]
```diff
@@ -103,7 +103,7 @@ secrecy = { version = "0.8.0", features = ["serde"] }
 serde = { version = "1.0.156", features = ["derive"] }
 serde_json = "1.0.97"
 sha2 = "0.10.6"
-socket2 = "0.4.9"
+socket2 = "0.5.7"
```
Comment (acmorrow) [inline Cargo.toml:106]: We shouldn't be upgrading socket2 here.

**Comment by acmorrow** [comment]
@mattjperez - I was able to get a very targeted update by doing the following:
```
diff --git a/Cargo.toml b/Cargo.toml
index 10e2c5fd..e1381243 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -70,7 +70,7 @@ futures-lite = "1"
 futures-rustls = "0.24"
 futures-util = "0.3.30"
 http-body-util = "0.1.1"
-hyper = { version = "1.2", default-features = false, features = ["server", "client", "http2"] }
+hyper = { version = "1.4", default-features = false, features = ["server", "client", "http2"] }
 ignore = "=0.4.20"
 lazy_static = "1.4.0"
 local-ip-address = "0.5.3"
```

And then:
```
$ cargo update hyper@1.3.1
```

The resulting total diff is:

```
diff --git a/Cargo.lock b/Cargo.lock
index a4075e98..45db9752 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -2070,9 +2070,9 @@ dependencies = [

 [[package]]
 name = "hyper"
-version = "1.3.1"
+version = "1.4.1"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "fe575dd17d0862a9a33781c8c4696a55c320909004a67a00fb286ba8b1bc496d"
+checksum = "50dfd22e0e76d0f662d429a5f80fcaf3855009297eab6a0a9f8543834744ba05"
 dependencies = [
  "bytes",
  "futures-channel",
@@ -2495,7 +2495,7 @@ dependencies = [
  "futures-rustls",
  "futures-util",
  "http-body-util",
- "hyper 1.3.1",
+ "hyper 1.4.1",
  "ignore",
  "lazy_static",
  "local-ip-address",
diff --git a/Cargo.toml b/Cargo.toml
index 10e2c5fd..e1381243 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -70,7 +70,7 @@ futures-lite = "1"
 futures-rustls = "0.24"
 futures-util = "0.3.30"
 http-body-util = "0.1.1"
-hyper = { version = "1.2", default-features = false, features = ["server", "client", "http2"] }
+hyper = { version = "1.4", default-features = false, features = ["server", "client", "http2"] }
 ignore = "=0.4.20"
 lazy_static = "1.4.0"
 local-ip-address = "0.5.3"
```

Which I think is what we want. Could you try that process in a clean tree and see if you reproduce my results? If so, I think that is what we should merge.

File: Cargo.toml [hunk 3]
```diff
@@ -70,7 +70,7 @@ futures-lite = "1"
 futures-rustls = "0.24"
 futures-util = "0.3.30"
 http-body-util = "0.1.1"
-hyper = { version = "1.2", default-features = false, features = ["server", "client", "http2"] }
+hyper = { version = "1.4.1", default-features = false, features = ["server", "client", "http2"] }
```
Comment (acmorrow) [inline Cargo.toml:73]: Is there a reason to specify `1.4.1` rather than `1.4`? The old requirement was just on `1.2`.

**Review by acmorrow — APPROVED** [review]
LGTM with one small question.

**Review by acmorrow — APPROVED** [review]
LGTM

**Pull request merged**